### PR TITLE
Improve support for internal and external loadbalancers

### PR DESCRIPTION
**Pull request by bruce-y**
Improves the usability of the internal loadbalancer pattern with correct usage examples.

### CODE REVIEW

File: examples/external-and-internal-lbs/main.tf [hunk 1]
```diff
@@ -12,20 +12,21 @@ module "internal_cert" {
 }
 
 module "internal_lb" {
-  source = "superblocksteam/superblocks/aws//modules/lb"
+  source  = "superblocksteam/superblocks/aws//modules/load-balancer"
```
Comment (taha-au) [inline examples/external-and-internal-lbs/main.tf:15]:
```suggestion
  source  = "superblocksteam/superblocks/aws/modules/load-balancer"
```

File: modules/load-balancer/variables.tf [hunk 2]
```diff
@@ -50,7 +55,8 @@ variable "create_dns" {
 }
 
 variable "zone_name" {
-  type = string
+  type    = string
+  default = "example.com"
```
Comment (taha-au) [inline modules/load-balancer/variables.tf:59]: don't add a default, add a description instead if the goal is to show format

File: variables.tf [hunk 3]
```diff
@@ -253,12 +253,20 @@ variable "certificate_arn" {
...
+}
+
 #################################################################
 # ECS
 #################################################################
-variable "lb_target_group_arn" {
-  type        = string
-  default     = null
+variable "lb_target_group_arns" {
+  type        = list(string)
```
Comment (taha-au) [inline variables.tf:268]: update description

File: variables.tf [hunk 4]
```diff
@@ -301,7 +309,7 @@ variable "create_ecs_sg" {
   description = "Whether to create default security group for ECS or not."
 }
 
-variable "load_balancer_sg_ids" {
+variable "allowed_load_balancer_sg_ids" {
```
Comment (taha-au) [inline variables.tf:312]: might wanna update description to the effect that these are additional on top of any created SGs

**Review by taha-au — DISMISSED** [review]

File: examples/external-and-internal-lbs/main.tf (hunk 1 above)
Comment (bruce-y) [reply examples/external-and-internal-lbs/main.tf:15]: Submodules are meant to be invoked with double slashes

https://registry.terraform.io/modules/terraform-aws-modules/rds/aws/1.0.3/submodules/db_instance

File: modules/load-balancer/variables.tf (hunk 2 above)
Comment (bruce-y) [reply modules/load-balancer/variables.tf:59]: default is needed because it could potentially not be provided if the dns name creation is set to false

Comment (taha-au) [reply modules/load-balancer/variables.tf:59]: id recommend setting default to null then

Comment (bruce-y) [reply modules/load-balancer/variables.tf:59]: works for me

**Review by taha-au — APPROVED** [review]

**Pull request merged**